tests/xdsl_opt/test_xdsl_opt.py

```diff
@@ -20,6 +20,7 @@ def test_opt():
         "stencil-shape-inference",
         "stencil-to-local-2d-horizontal",
         "frontend-desymrefy",
+        "dce",
```

Comment:
I remember us discussing a more insightful name for this (deadcodeelimination or something) as it was brought up that non compiler people might be looking at this in the future. Maybe this could be solved be some docu in the tutorials? (Not for this PR)

Reply:
I guess they might count it as cononicalization, which it is somewhat...

I knwo that LLVM has a dce and an a(ggressive) dce pass, so I guess this is not fully agreed upon.